test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -618,7 +619,11 @@ protected RestClient buildClient(Settings settings, HttpHost[] hosts) throws IOE
     /**
      * Whether the used REST client should return any response containing at
      * least one warning header as a failure.
+     * @deprecated always run in strict mode and use
+     *   {@link RequestOptions.Builder#setWarningsHandler} to override this
+     *   behavior on individual requests
      */
+    @Deprecated
     protected boolean getStrictDeprecationMode() {
```

Comment:
Got it. Now I see the relationship between WarningsHandler and setStrictDeprecationMode. Maybe make that earlier Javadoc more explicit about the fact setStrictDeprecationMode has been deprecated?

Reply:
I don't think RestClientBuilder#setStrictDeprecationMode has been deprecated in this PR but maybe it should be. Or we could do it as a followup.

Reply:
I don't feel to bad about having strict deprecation mode on the builder and this more expressive WarningsHandler thing on the RequestOptions. I don't think setting a WarningsHandler on the builder makes sense because it is about being careful and you can't really be careful for *all* of the requests that a client makes.